src/utilities/delete-collection.tsx

```diff
@@ -0,0 +1,191 @@
+import { CollectionAPI } from 'src/api';
+
+import React from 'react';
+import { errorMessage, parsePulpIDFromURL, waitForTask } from 'src/utilities';
+import { t, Trans } from '@lingui/macro';
+import { Paths, formatPath } from 'src/paths';
+
+import { Text, DropdownItem, Checkbox, Tooltip } from '@patternfly/react-core';
+
+import { DeleteModal } from 'src/components';
+
+class DeleteCollectionUtils {
+  public getUsedbyDependencies(collection, setDependencies, setAlert) {
+    const { name, namespace } = collection;
+    CollectionAPI.getUsedDependenciesByCollection(namespace.name, name)
+      .then(({ data }) => {
+        setDependencies(!data.data.length);
+      })
+      .catch((err) => {
+        const { status, statusText } = err.response;
+        setAlert({
+          variant: 'danger',
+          title: t`Dependencies for collection "${name}" could not be displayed.`,
+          description: errorMessage(status, statusText),
+        });
+      });
+  }
+
+  public deleteMenuOption(
+    noDependencies,
+    delete_collection_permission: boolean,
+    onClick,
+  ) {
+    let ret = noDependencies
+      ? delete_collection_permission && (
+          <DropdownItem
+            key='delete-collection-enabled'
+            onClick={() => onClick()}
+            data-cy='delete-collection-dropdown'
+          >
+            {t`Delete entire collection`}
+          </DropdownItem>
+        )
+      : delete_collection_permission && (
+          <Tooltip
+            key='delete-collection-disabled'
+            position='left'
+            content={
+              <Trans>
+                Cannot delete until collections <br />
+                that depend on this collection <br />
+                have been deleted.
+              </Trans>
+            }
+          >
+            <DropdownItem isDisabled>
+              {t`Delete entire collection`}
+            </DropdownItem>
+          </Tooltip>
+        );
+
+    if (!ret) {
+      ret = <div></div>;
+    }
+    return ret;
```

Comment:
can we simplify the logic by returning early?

```suggestion
    if (!delete_collection_permission) {
      return <div></div>;
    }
    
    return noDependencies ? (
          <DropdownItem
            key='delete-collection-enabled'
            onClick={() => onClick()}
            data-cy='delete-collection-dropdown'
          >
            {t`Delete entire collection`}
          </DropdownItem>
        ) : (
          <Tooltip
            key='delete-collection-disabled'
            position='left'
            content={
              <Trans>
                Cannot delete until collections <br />
                that depend on this collection <br />
                have been deleted.
              </Trans>
            }
          >
            <DropdownItem isDisabled>
              {t`Delete entire collection`}
            </DropdownItem>
          </Tooltip>
        );
```

Reply:
also .. wasn't `<div></div>` problematic inside menu options? Should this return null instead?

(adding a padding-sized item)

Reply:
The null threw the errors and it works with div.

Reply:
Yeah, because you were missing those `filter(Boolean)` bits before.

Again, we can't use empty divs here, it has to be falsy